XenAdmin/Wizards/BallooningWizard_Pages/MemorySettings.cs

```diff
@@ -64,7 +65,11 @@ protected override void PageLoadedCore(PageLoadedDirection direction)
             if (!_changed)
                 return;
 
-            if (VMs != null && VMs.Count > 0 && VMs[0].advanced_ballooning())
+            var connection = VMs?.FirstOrDefault()?.Connection;
+            var ballooningInUse = VMs?.FirstOrDefault()?.UsesBallooning();

```

Comment:
Do we not need to check if Any VMs have `UsesBallooning` set?

Reply:
FirstOrDefault() is essentially the same as Any(), the difference being that it also returns the first item found rather than just that it has been found. But I suppose Any() makes for more elegant code here - I'll change it 👍

Reply:
I've just realised that my comment above was not relevant to what you meant. My reply was about `VMs?.FirstOrDefault(v=>v.UsesBallooning())` vs `VMs?.Any(v=>v.UsesBallooning())`. 

`VMs?.FirstOrDefault()?.UsesBallooning()` is slightly better that the original `VMs[0].UsesBallooning` code, but I'm afraid I don't know why the original code was only checking for the first VM, I agree that `VMs?.Any(v=>v.UsesBallooning())` makes more sense.

Reply:
It is likely that `VMs?.All(v=>v.UsesBallooning())` is the most correct answer of all. But given the uncertainty that exists in this area, I think I'll leave it as is for now because it is closer to the original code. I'll revisit this within CP-39141, because this correction is outside the scope of CP-39142.